### PR TITLE
Fixes cannot disable file auto save when configuration target is other than user

### LINKED ISSUE #108932: Cannot disable file auto save
I cannot disable file auto save.

**I cannot uncheck "Auto Save" under "File" menu.** I disabled it in User settings but this option is still checked in file menu.   

Restarting VS Code has no effect. I tried running VS Code without extensions and it behaves exactly the same. So it's very likely not related to any extension.

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #108932

### CODE REVIEW

**Comment by bpasero** [comment]
@jeanp413 thanks, this change made me look closer into the `configurationService.updateValue` method. As you can see, it seems to already do exactly what you do manually from within the implementation when you leave out the target:

https://github.com/microsoft/vscode/blob/8fcf778ba251131780f2a110d311161f39b97b03/src/vs/workbench/services/configuration/browser/configurationService.ts#L710-L735

I actually think that we should not have code outside the configuration service that tries to find the right target, it should be handled consistently within the configuration service.

@sandy081 need your advice here: searching for references to `ConfigurationTarget.USER` I see a ton of uses, e.g. in the editor to toggle things, such as minimap:

https://github.com/microsoft/vscode/blob/f3454533fb3403f1668897e5799a71be3a260a8a/src/vs/workbench/contrib/codeEditor/browser/toggleMinimap.ts#L26-L29

If a user has configured the `editor.minimap.enabled` setting as a workspace setting, these actions actually do nothing. 

To summarise, I think the auto save toggle is just the tip of the ice berg, any `updateValue` with `ConfigurationTarget.USER` will fail unless the setting is not allowed per workspace.

**Comment by jeanp413** [comment]
Removed the duplicated code, let me know if fixing the other places that you pointed out will be part of this pr or will be tackle apart.

**Comment by bpasero** [comment]
Thanks, I have extracted https://github.com/microsoft/vscode/issues/109373 for follow up on the other locations.

**Comment by sandy081** [comment]
As a core service `IConfigurationService` provides multiple options for the consumer to update - like updating in specific target or resource etc., Its up to the caller to use the variant it needed. I agree that most of the toggle actions does not need to send the target and just use the `updateValue(key, value)` and the service does the right thing.
